### ISSUE TITLE
Expired session can be revived

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3458]
|[<img alt="mthuurne's avatar" src="https://avatars.githubusercontent.com/u/246676?s=50" width="50" height="50">](https://github.com/mthuurne)| @mthuurne reported|
|-|-|
|Trac ID|trac#3458|
|Type|defect|
|Created|2008-10-03 10:26:00Z|

The Request.getSession() method does not check whether an existing session has already expired. Also, it calls Session.touch(), so if the session was expired (past its sessionTimeout) but was not cleaned up yet (before the sessionCheckTime), it will be revived.

The time between session expiry and session cleanup can be at most (sessionCheckTime - sessionTimeout) seconds, which is 15 minutes with the default values. Fixing [#3457](https://github.com/twisted/twisted/issues/3457) would reduce the delay between expiry and cleanup, but may or may not eliminate it, depending on the strategy chosen. If this delay is eliminated, the expired session revival could not occur anymore, otherwise it would require a separate bugfix.

One solution would be to have Request.getSession() treat expired sessions like they don't exist and create a new session if the previous one expired. Another solution would be to have Site.getSession(), which is used by Request.getSession(), return only non-expired sessions. I prefer the latter, since it prevents expired sessions from being used in other places besides Request.

<details><summary>Searchable metadata</summary>

```
trac-id__3458 3458
type__defect defect
reporter__mthuurne mthuurne
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1223029560000000 1223029560000000
changetime__1233182951000000 1233182951000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

The fix for [#3457](https://github.com/twisted/twisted/issues/3457), r26133, removed the possibility for this.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
